libraries/botbuilder-dialogs/tests/memory_memoryScopes.test.js

```diff
@@ -488,6 +490,76 @@ describe('Memory - Memory Scopes', function () {
             'two',
             'settings should be overriden by environment variables'
         );
+        */
+    });
+
+    it('SettingsMemoryScope should get settings from configuration.', async function () {
+        // Create a DialogState property, DialogSet and register the dialogs.
+        const convoState = new ConversationState(new MemoryStorage());
+        const dialogState = convoState.createProperty('dialogs');
+        const dialogs = new DialogSet(dialogState).add(new TestDialog('test', 'test message'));
+
+        // Create test context
+        const context = new TurnContext(new TestAdapter(), beginMessage);
+        const dc = await dialogs.createContext(context);
+        const configuration = {
+            simple: 'test',
+            'object:simple': 'test',
+            'array:0': 'one',
+            'array:1': 'two',
+            'object:array:0': 'one',
+            'object:array:1': 'two',
+        };
+        dc.context.turnState.set(DialogTurnStateConstants.configuration, configuration);
+
+        // Run test
+        const scope = new SettingsMemoryScope();
+        const memory = scope.getMemory(dc);
+        assert.strictEqual(typeof memory, 'object', `settings not returned`);
```

Comment:
Check out `assert.deepStrictEqual`, it can make these assertions a bit more natural.

Reply:
I tried `assert.deepStrictEqual` but failed because there are some environment variables be filled into memory so we cannot compare it with the test configuration.